### ISSUE TITLE
Should warn about useless `exit 0`

### ISSUE THREAD

**Issue by zapashcanon** [koalaman/shellcheck#952]
Hi,

- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/sh

# anything

exit 0

```

#### Here's what shellcheck currently says:

Nothing.

#### Here's what I wanted or expected to see:

`exit 0` is useless at the end of a script

**Comment by knirch** [comment]
The exit status of the script will be the exit status of the last command.

While mostly it's IMHO cleaner to handle unwanted exit status to with 
command || true an exit 0 is also a way to convey that the script reached a 
good exit point.

Many scripts that have a big if our case with multiple exit points want to 
have a exit 0 at the end, just to make it clearer for the reader that all 
is good.


Hi,

- [x] shellcheck.net (i.e. the latest commit) currently gives no useful 
warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and 
didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/sh

# anything

exit 0

```

#### Here's what shellcheck currently says:

Nothing.

#### Here's what I wanted or expected to see:

`exit 0` is useless at the end of a script


-- 
You are receiving this because you are subscribed to this thread.
Reply to this email directly or view it on GitHub:
https://github.com/koalaman/shellcheck/issues/952

**Comment by zapashcanon** [comment]
> While mostly it's IMHO cleaner to handle unwanted exit status to with command || true an exit 0 is also a way to convey that the script reached a good exit point.

Then, we should add a warning and tell the user to use `command || true` instead of `command; exit 0`.

What do you think about the case where `set -e` is present ? IMO, in this case, there's no point at all in adding `exit 0`, so, we should warn the user for sure here.
